### PR TITLE
Refactor upgrades/test_foreman_maintain for pytest

### PR DESCRIPTION
**Pull request by Gauravtalreja1**
**Test Results:**

   **Pre-Upgrade:**
```
(.robottelo_py38) [gtalreja@gtalreja robottelo]$ pytest -m pre_upgrade tests/upgrades/test_foreman_maintain.py 
===================================================== test session starts =====================================================
platform linux -- Python 3.8.2, pytest-6.2.2, py-1.9.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/gtalreja/sat_workspace/robottelo, configfile: pyproject.toml
plugins: mock-3.5.1, xdist-2.2.1, cov-2.11.1, ibutsu-1.14.1, services-2.2.1, forked-1.3.0, reportportal-5.0.8
collected 2 items / 1 deselected / 1 selected                                                                                 

tests/upgrades/test_foreman_maintain.py .                                                                               [100%]

====================================================== warnings summary =======================================================
<SOME_WARNINGS />
======================================== 1 passed, 1 deselected, 3 warnings in 24.90s =========================================
2021-03-19 14:49:54 - robottelo - INFO - Save failed tests to file /var/tmp/robottelo_pre_upgrade_failed_tests.json
```

  **Post-Upgrade:**
```
(.robottelo_py38) [gtalreja@gtalreja robottelo]$ pytest -m post_upgrade tests/upgrades/test_foreman_maintain.py 
===================================================== test session starts =====================================================
platform linux -- Python 3.8.2, pytest-6.2.2, py-1.9.0, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/gtalreja/sat_workspace/robottelo, configfile: pyproject.toml
plugins: mock-3.5.1, xdist-2.2.1, cov-2.11.1, ibutsu-1.14.1, services-2.2.1, forked-1.3.0, reportportal-5.0.8
collected 2 items / 1 deselected / 1 selected                                                                                 

tests/upgrades/test_foreman_maintain.py .                                                                               [100%]

====================================================== warnings summary =======================================================
<SOME_WARNINGS />
======================================== 1 passed, 1 deselected, 3 warnings in 24.77s =========================================
```
Signed-off-by: Gaurav Talreja <gtalreja@redhat.com>

### CODE REVIEW

File: tests/upgrades/test_foreman_maintain.py [hunk 1]
```diff
@@ -16,25 +16,21 @@
...
 """
-from unittest2.case import TestCase
 from upgrade_tests import post_upgrade
 from upgrade_tests import pre_upgrade
 
 from robottelo import ssh
 
 
-class ScenarioForemanMaintain(TestCase):
+class TestForemanMaintain:
```
Comment (JacobCallahan) [inline tests/upgrades/test_foreman_maintain.py:25]: Is there a need to keep the class? If not, you can remove it and make all tests module-level

**Review by JacobCallahan — CHANGES_REQUESTED** [review]

File: tests/upgrades/test_foreman_maintain.py (hunk 1 above)
Comment (Gauravtalreja1) [reply tests/upgrades/test_foreman_maintain.py:25]: Yes, it's kinda doable but here with this PR trying to keep the same structure for all tests under `/tests/upgrades`, like keeping a class, and tests inside a class.

Comment (JacobCallahan) [reply tests/upgrades/test_foreman_maintain.py:25]: I later saw a discussion where I believe @jyejare said classes were required for upgrade tests.

We can leave as-is for now, but I'd like to eventually come back around and address that limitation.

**Review by JacobCallahan — APPROVED** [review]

**Review by omkarkhatavkar — APPROVED** [review]
ACK

**Pull request merged**